geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedFeatureSourceView.scala

```diff
@@ -0,0 +1,172 @@
+    val bounds = new ReferencedEnvelope(org.locationtech.geomesa.utils.geotools.CRS_EPSG_4326)
+    sources.foreach(s => bounds.expandToInclude(s.getBounds(query)))
+    bounds
+  }
+
+  override def getQueryCapabilities: QueryCapabilities = capabilities
+
+  override def getFeatures: SimpleFeatureCollection = getFeatures(Filter.INCLUDE)
+
+  override def getFeatures(filter: Filter): SimpleFeatureCollection = getFeatures(new Query(sft.getTypeName, filter))
+
+  override def getFeatures(query: Query): SimpleFeatureCollection = new MergedFeatureCollection(query)
+
+  override def getName: Name = getSchema.getName
+
+  override def getDataStore: DataStore = ds
+
+  override def getSupportedHints: java.util.Set[Key] = hints
+
+  override def getInfo: ResourceInfo = new DelegatingResourceInfo(this)
+
+  override def addFeatureListener(listener: FeatureListener): Unit = throw new NotImplementedError()
+
+  override def removeFeatureListener(listener: FeatureListener): Unit = throw new NotImplementedError()
+
+  /**
+    * Feature collection implementation
+    *
+    * @param query query
+    */
+  class MergedFeatureCollection(query: Query) extends DataFeatureCollection(GeoMesaFeatureCollection.nextId) {
+
+    private val open = new AtomicBoolean(false)
+
+    override def getSchema: SimpleFeatureType = {
+      if (!open.get) {
+        // once opened the query will already be configured by the query planner,
+        // otherwise we have to compute it here
+        ds.runner.configureQuery(sft, query)
+      }
+      // copy the query so that we don't set any hints for transforms, etc
+      val copy = new Query(query)
+      copy.setHints(new Hints(query.getHints))
+      QueryPlanner.setQueryTransforms(copy, sft)
+      ds.runner.getReturnSft(sft, copy.getHints)
+    }
+
+    override protected def openIterator(): java.util.Iterator[SimpleFeature] = {
+      val iter = super.openIterator()
+      open.set(true)
+      iter
+    }
+
+    override def accepts(visitor: FeatureVisitor, progress: ProgressListener): Unit =
+      visitor match {
+        case v: BoundsVisitor =>
+          v.reset(ds.stats.getBounds(sft, query.getFilter))
+
+        case v: MinVisitor if v.getExpression.isInstanceOf[PropertyName] =>
+          val attribute = v.getExpression.asInstanceOf[PropertyName].getPropertyName
+          minMax(attribute, exact = false).orElse(minMax(attribute, exact = true)) match {
+            case Some((min, _)) => v.setValue(min)
+            case None           => super.accepts(visitor, progress)
+          }
+
+        case v: MaxVisitor if v.getExpression.isInstanceOf[PropertyName] =>
+          val attribute = v.getExpression.asInstanceOf[PropertyName].getPropertyName
+          minMax(attribute, exact = false).orElse(minMax(attribute, exact = true)) match {
+            case Some((_, max)) => v.setValue(max)
+            case None           => super.accepts(visitor, progress)
+          }
+
+        case v: GeoMesaProcessVisitor =>
```

Comment:
Hmm...  How are GeoMesa Processes going to work on Merged Views?  That seems like it could be complicated at best...  Do we have any unit tests for that?

Reply:
i don't have any tests for that specifically, but I have tests for the underlying query types that the process visitors invoke. Generally, they set some query hints (e.g. ARROW_SCAN = true) and call the underlying data store. The [MergedQueryRunner](https://github.com/locationtech/geomesa/pull/2088/files#diff-8195d6160bc8ed355726a3a9f0cae929R44) handles merging the results from the underlying stores, depending on if they handle the hints natively (like geomesa stores do) or else it will handle it client side (for e.g. postgis).
See https://github.com/locationtech/geomesa/pull/2088/files#diff-dcd9b3c4473e849d494e0bdf8b59673cR204 for tests